src/test/java/in/zapr/druid/druidry/query/aggregation/MovingAverageTest.java

```diff
@@ -0,0 +1,179 @@
+
+        String actualJson = objectMapper.writeValueAsString(query);
+        JSONAssert.assertEquals(actualJson, expectedJson, JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+    @DataProvider(name = "averagerTypesProvider")
+    private static Object[][] averagerTypesProvider() {
+        return new Object[][]{
+                {"doubleMax", (MovingAverageCreators.Creator) MovingAverageCreators::doubleMax},
+                {"doubleMean", (MovingAverageCreators.Creator) MovingAverageCreators::doubleMean},
+                {"doubleMeanNoNulls", (MovingAverageCreators.Creator) MovingAverageCreators::doubleMeanNoNulls},
+                {"doubleMin", (MovingAverageCreators.Creator) MovingAverageCreators::doubleMin},
+                {"doubleSum", (MovingAverageCreators.Creator) MovingAverageCreators::doubleSum},
+                {"longMax", (MovingAverageCreators.Creator) MovingAverageCreators::longMax},
+                {"longMean", (MovingAverageCreators.Creator) MovingAverageCreators::longMean},
+                {"longMeanNoNulls", (MovingAverageCreators.Creator) MovingAverageCreators::longMeanNoNulls},
+                {"longMin", (MovingAverageCreators.Creator) MovingAverageCreators::longMin},
+                {"longSum", (MovingAverageCreators.Creator) MovingAverageCreators::longSum}
+        };
+    }
+
+    private String loadExpectedJsonForType(String type) throws IOException, URISyntaxException {
+        String json = loadExpectedJson("averager_type/" + type + ".json");
+        assertTrue(json.contains(String.format("\"type\": \"%s\"", type)));
+        return json;
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    private String loadExpectedJson(String fileName) throws IOException, URISyntaxException {
+        URI jsonUri = Resources.getResource("query/aggregation/moving_average/" + fileName).toURI();
+        File jsonFile = new File(jsonUri);
+        return Files.readFile(jsonFile);
+    }
+
+    private static DruidMovingAverageQuery simpleQuery(
+            MovingAverageCreators.Creator averagerCreator) {
+        DateTime startTime = new DateTime(2020, 1, 1, 0, 0, 0, DateTimeZone.UTC);
+        DateTime endTime = new DateTime(2020, 2, 29, 23, 59, 59, DateTimeZone.UTC);
+        Interval interval = new Interval(startTime, endTime);
+        String averagedName = "averagedName";
+        DruidAggregator aggregator = new DoubleSumAggregator(averagedName, "aggregatedFieldName");
+        DruidAverager averager = averagerCreator.apply("averagingResult", averagedName, 31);
+
+        return DruidMovingAverageQuery.builder()
+                .dataSource(new TableDataSource("simpleDataSource"))
+                .granularity(new SimpleGranularity(PredefinedGranularity.DAY))
+                .aggregations(singletonList(aggregator))
+                .intervals(singletonList(interval))
+                .averagers(singletonList(averager))
+                .build();
+    }
+
```

Comment:
Could you add test for not building with required params and `expectedExceptions = NullPointerException.class`. Basically testing for required fields missing.